text/0757-default-modifier-manager.md

```diff
@@ -0,0 +1,276 @@
+class A {
+  myModifier = (element: Element, first: number, second: number, options: Options) => {
+    // ...
+  }
+}
+```
+
+#### Example with only positional parameters
+
+```hbs
+<div {{this.myModifier 1 2}}></div>
+```
+Because there are no named arguments passed in, the develope may opt to omit the options argument from the method signature.
+```ts
+class A {
+  myModifier = (element: Element, first: number, second: number) => {
+    // ..
+  }
+}
+```
+however, the options argument is still passed, but is equivalent to `{}`.
+
+
+#### Example default modifier implementation
+
+The type signature for any function-modifier could be:
+
+```ts
+import type { CapturedArguments as Arguments } from '@glimmer/interfaces';
+
+type FnArgs<Args extends Arguments = Arguments> =
+  | [Element, ...Args['positional'], Args['named']]
+  | [Element, ...Args['positional'], {}];
+
+type FunctionModifier<
+  Args extends Arguments = Arguments,
+  ModifierArgs extends FnArgs<Args> = FnArgs<Args>
+> =
+  | ((...args: ModifierArgs) => void) // setup, with no teardown
+  | ((...args: ModifierArgs) => () => void) // setup with teardown
+
+class A {
+  myModifierA: FunctionModifier = (element, a, b, c, options) => {
+    /* setup, with no teardown */
+  }
+  myModifierB: FunctionModifier = (element, a, b, c, options) => {
+    /* setup */
+    return () => { /* teardown */ }
+  }
+}
+```
+
+
+The details of the implementation of this modifier manager may take inspiration from
+ - The [implementation](gh-vm-1348) for [RFC #756](rfc-756)
+   (especially around where the default helper manager is chosen as the "fallback" when an existing manager is not found)
+ - The modifier-manager in [ember-could-get-used-to-this](ecgutt-modifier)
+   The main differences being that this RFC proposes
+   - no need to import anything or wrap in a function call
+   - instead of `(element, positionalArgs, namedArgs)` as the function signature, the signature
+     will be `(element, ...positionalArgs, namedArgs)`
+
+[gh-vm-1348]: https://github.com/glimmerjs/glimmer-vm/pull/1348
+[rfc-756]: https://github.com/emberjs/rfcs/pull/756
+[ecgutt-modifier]: https://github.com/tracked-tools/ember-could-get-used-to-this/blob/master/addon/-private/modifiers.js
+
+## How we teach this
+
+On the [Template Lifecycle, DOM, and Modifiers](tldm) page,
+we'll want to insert a section on custom modifiers towards the end, and maybe swap out the
+existing documentation referencing `ember-modifier` and/or `@ember/render-modifiers`.
+We'll also want examples of how to manage modifiers with tracked data, maybe when to use a
+local/private modifier, rather than something you'd put in `app/modifiers`.
+Maybe a whole page is added to the guides on modifiers, their philosophy, when to use them, etc.
+
+While the overall API for modifiers in general has been "not perfect", there has been a fair amount of
+experimentation with modifiers since the 3.11 - 3.16 (pre-octane) era. Function modifiers are
```

Comment:
I'd also note here that there has also been extensive usage of `ember-modifier` *since* the start of the Octane era, which has proven that the function-based modifier design is solid for its use cases, and that all the open questions are specifically around class-based modifiers with their lifecycle management.

Reply:
added!